src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -583,7 +586,7 @@ ParquetBlockInputFormat::ParquetBlockInputFormat(
 {
     if (max_decoding_threads > 1)
         pool = std::make_unique<ThreadPool>(CurrentMetrics::ParquetDecoderThreads, CurrentMetrics::ParquetDecoderThreadsActive, CurrentMetrics::ParquetDecoderThreadsScheduled, max_decoding_threads);
-    if (supportPrefetch())
+    if (supportPrefetch() || format_settings.parquet.use_native_reader_with_filter_push_down)
         io_pool = std::make_shared<ThreadPool>(CurrentMetrics::ParquetDecoderIOThreads, CurrentMetrics::ParquetDecoderIOThreadsActive, CurrentMetrics::ParquetDecoderIOThreadsScheduled, max_io_threads);
```

Comment:
Missing `io_pool->wait()` in `resetParser()`. `RowGroupPrefetch`'s `task` may use-after-free without it.